### PR TITLE
CBG-3522 add collection to import feed logic

### PR DESCRIPTION
**Pull request by torcolvin**
## Pre-review checklist
- [x] Removed debug logging (`fmt.Print`, `log.Print`, ...)
- [x] Logging sensitive data? Make sure it's tagged (e.g. `base.UD(docID)`, `base.MD(dbName)`)
- [x] Updated relevant information in the API specifications (such as endpoint descriptions, schemas, ...) in `docs/api`

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [x] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/2098/

### CODE REVIEW

File: db/import_listener.go [hunk 1]
```diff
@@ -130,35 +130,43 @@ func (il *importListener) StartImportFeed(dbContext *DatabaseContext) (err error
...
 		return true
 	}
 
-	il.ImportFeedEvent(event)
+	il.ImportFeedEvent(ctx, collection, event)
 	return true
 }
 
-func (il *importListener) ImportFeedEvent(event sgbucket.FeedEvent) {
+func (il *importListener) ImportFeedEvent(ctx context.Context, collection DatabaseCollectionWithUser, event sgbucket.FeedEvent) {
```
Comment (bbrks) [inline db/import_listener.go]: As you've noted, I think this does need to be a pointer to allow `ReloadUser` to change the user inside this struct - I'm not sure whether this function is called from here or not.

Comment (torcolvin) [reply db/import_listener.go]: The part that is weird to me is that this was _never_ a pointer and I didn't change this when migrating to collections https://github.com/couchbase/sync_gateway/commit/dd9e90cd2cfdfed0f4568143d9641d751126ef83

Comment (torcolvin) [reply db/import_listener.go]: I think user doesn't matter because there's no user associated with import feed.

File: db/import_listener.go [hunk 2]
```diff
@@ -130,35 +130,43 @@ func (il *importListener) StartImportFeed(dbContext *DatabaseContext) (err error
...
 	if event.Opcode != sgbucket.FeedOpMutation && event.Opcode != sgbucket.FeedOpDeletion {
 		return true
 	}
-	key := string(event.Key)
+	docID := string(event.Key)
+
+	collection, ok := il.collections[event.CollectionID]
+	if !ok {
+		base.WarnfCtx(ctx, "Received import event for unrecognised collection 0x%x", event.CollectionID)
+		return
```
Comment (adamcfraser) [inline db/import_listener.go]:
```suggestion
		return true
```
Looks like this should return true to preserve the previous handling.

File: db/import_listener.go (hunk 1 above)
Comment (adamcfraser) [reply db/import_listener.go]: Should this be a pointer in any case, to avoid any unwanted copying?

**Review by adamcfraser — CHANGES_REQUESTED** [review]

File: db/import_listener.go (hunk 1 above)
Comment (torcolvin) [reply db/import_listener.go]: Made a ptr.

**Review by adamcfraser — APPROVED** [review]

**Pull request merged**